### DISCUSSION TITLE
How to add a custom dialogue

### DISCUSSION THREAD

**Discussion by Reyororo** [KadeArchive/Kade-Engine#1687] [Q&A] [answered]
How do I add a custom dialogue?

**Comment by raf13lol** [comment, marked as answer]
on playstate.hx and dialoguebox.hx edit some code to make it work

**Reply by Reyororo** [reply]
> on playstate.hx and dialoguebox.hx edit some code to make it work

please tell me in detail

**Reply by raf13lol** [reply]
dialoguebox.hx allows for portraits and playstate.hx handles if there is dialogue

**Reply by Reyororo** [reply]
> dialoguebox.hx allows for portraits and playstate.hx handles if there is dialogue

Please tell me the code etc. in detail

**Reply by raf13lol** [reply]
ok so dialoguebox.hx  at line 67

//this whole thing is on the latest master code
case 'insert song name here':
				hasDialog = true;
				//this plays sound so yeah
                                 FlxG.sound.play(Paths.sound('ANGRY_TEXT_BOX')); 
                                //this controls what the box looks like
      				box.frames = Paths.getSparrowAtlas('weeb/pixelUI/dialogueBox-senpaiMad');
				//this plays any animation for the box via the xml
                                box.animation.addByPrefix('normalOpen', 'SENPAI ANGRY IMPACT SPEECH', 24, false);
				box.animation.addByIndices('normal', 'SENPAI ANGRY IMPACT SPEECH', [4], "", 24);

to add the portrait umm idk just ask for help on discord if needed because i dont know openfl but i know how to copy and paste and it works so yeah

now that you are on playstate.hx go to line 493

case 'senpai' | 'roses' | 'thorns' | 'insert song name here':

if you want multiple songs to have dialogue

case 'senpai' | 'roses' | 'thorns' | 'insert song 1 name here' | 'insert song 2 name here' | 'insert song 3 name here' :

then put dialogue in the song folders as dialogue.txt

Hope This Helps! :)
